BUILD.bazel

```diff
@@ -980,25 +937,215 @@ ray_cc_library(
+        "//src/ray/protobuf:node_manager_cc_proto",
+    ],
+)
+
+ray_cc_library(
+    name = "cluster_resource_manager",
+    srcs = ["src/ray/raylet/scheduling/cluster_resource_manager.cc"],
+    hdrs = ["src/ray/raylet/scheduling/cluster_resource_manager.h"],
+    deps = [
+        ":local_resource_manager",
+        ":ray_common",
+        "@com_google_absl//absl/container:flat_hash_map",
+        "@com_google_googletest//:gtest",
+    ],
+)
+
+ray_cc_library(
+    name = "cluster_resource_scheduler",
+    srcs = [
+        "src/ray/raylet/scheduling/cluster_resource_scheduler.cc",
+    ],
+    hdrs = [
+        "src/ray/raylet/scheduling/cluster_resource_scheduler.h",
+    ],
+    deps = [
+        ":cluster_resource_manager",
+        ":composite_scheduling_policy",
+        ":scheduler_internal",
+    ],
+)
+
+ray_cc_library(
+    name = "cluster_task_manager",
+    srcs = [
+        "src/ray/raylet/scheduling/cluster_task_manager.cc",
+        "src/ray/raylet/scheduling/scheduler_stats.cc",
+    ],
+    hdrs = [
+        "src/ray/raylet/scheduling/cluster_task_manager.h",
+        "src/ray/raylet/scheduling/scheduler_stats.h",
+    ],
+    deps = [
+        ":cluster_resource_scheduler",
+        ":cluster_task_manager_interface",
+        ":local_task_manager_interface",
+        ":scheduler_resource_reporter",
+    ],
+)
+
+ray_cc_library(
+    name = "cluster_task_manager_interface",
+    hdrs = ["src/ray/raylet/scheduling/cluster_task_manager_interface.h"],
+)
+
+ray_cc_library(
+    name = "local_task_manager_interface",
+    hdrs = ["src/ray/raylet/scheduling/local_task_manager_interface.h"],
+)
+
+ray_cc_library(
+    name = "local_resource_manager",
+    srcs = ["src/ray/raylet/scheduling/local_resource_manager.cc"],
+    hdrs = ["src/ray/raylet/scheduling/local_resource_manager.h"],
     deps = [
         ":gcs_client_lib",
         ":ray_common",
-        ":stats_lib",
-        "//src/ray/protobuf:common_cc_proto",
-        "//src/ray/util",
-        "@boost//:asio",
-        "@boost//:system",
-        "@com_github_jupp0r_prometheus_cpp//pull",
-        "@com_google_absl//absl/base:core_headers",
+        "@com_google_absl//absl/container:flat_hash_map",
         "@com_google_absl//absl/container:flat_hash_set",
-        "@com_google_absl//absl/memory",
-        "@com_google_absl//absl/random",
-        "@com_google_absl//absl/random:bit_gen_ref",
-        "@com_google_absl//absl/strings",
         "@com_google_googletest//:gtest",
-        "@io_opencensus_cpp//opencensus/exporters/stats/prometheus:prometheus_exporter",
-        "@io_opencensus_cpp//opencensus/stats",
-        "@io_opencensus_cpp//opencensus/tags",
+    ],
+)
+
+ray_cc_library(
+    name = "affinity_with_bundle_scheduling_policy",
+    srcs = ["src/ray/raylet/scheduling/policy/affinity_with_bundle_scheduling_policy.cc"],
+    hdrs = ["src/ray/raylet/scheduling/policy/affinity_with_bundle_scheduling_policy.h"],
+    deps = [":scheduling_policy"],
```

Comment:
could you please also add 
https://github.com/ray-project/ray/blob/1103bd114c3dcc4235d0fa8cbbd5b89f7d00dc99/src/ray/raylet/scheduling/policy/affinity_with_bundle_scheduling_policy.h#L17